hls/decode-util.go

```diff
@@ -113,13 +113,18 @@ func decodeSessionData(line string) *SessionData {
 	return sd
 }
 
-func decodeInf(line string) (*Inf, error) {
+func decodeInf(line string, version int) (*Inf, error) {
 	var err error
 	i := &Inf{}
 	index := strings.Index(line, ",")
 	if i.Duration, err = strconv.ParseFloat(line[0:index], 64); err != nil {
 		return nil, err
 	}
+	if version < 3 {
+		if i.Duration != float64(int64(i.Duration)) {
```

Comment:
I checked on the math here.. doing it this way is ok as long as you are below 2^63 for the integer component of the float... considering we likely won't go above 10... figured it was ok to do the conversion like so rather than using math.Trunc.